lib/chef/resource/windows_share.rb

```diff
@@ -338,8 +335,38 @@ def bool_string(bool)
           # bool ? 1 : 0
           bool ? "$true" : "$false"
         end
+
+        # Grant-SmbShareAccess sets only one permission to one user so we need to remove from the lower permission access
+        # For change_users remove common full_users from it
+        # For read_users remove common full_users as well as change_users
+        def new_resource_users
+          users = {}
+          users["full_users"] = new_resource.full_users
+
+          users["change_users"] = new_resource.change_users - new_resource.full_users
+          users["read_users"] = new_resource.read_users - (new_resource.full_users + new_resource.change_users)
+          users
+        end
+
+        # Compare the full_users, change_users and read_users from current_resource and new_resource
+        # @returns boolean True/False
+        def compare_users
```

Comment:
This should be named something like `def users_changed?` and the sense of it should be inverted (this makes it read nicer without the NOT when it gets used above)